packages/clerk-js/src/core/resources/SignUp.ts

```diff
@@ -158,10 +158,12 @@ export class SignUp extends BaseResource implements SignUpResource {
     return this.attemptVerification({ signature: generatedSignature, strategy: 'web3_metamask_signature' });
   };
 
-  public authenticateWithWeb3 = async (params: AuthenticateWithWeb3Params): Promise<SignUpResource> => {
-    const { generateSignature, identifier } = params || {};
+  public authenticateWithWeb3 = async (
+    params: AuthenticateWithWeb3Params & { unsafeMetadata: SignUpUnsafeMetadata },
```

Comment:
❓ Can we add `unsafeMetadata` to the `AuthenticateWithWeb3Params` type directly?

Reply:
No as this is a shared type between the SignIn and the SignUp object. The `unsafeMetadata` are only available on the signUp object.